notify failing on MacOS with new `fsevent-sys` update

#### System details

<!-- Please include ALL of the following: -->

- OS/Platform name and version: MacOS - latest
- Rust version (if building from source): `rustc --version`: 1.52
- Notify version (or commit hash if building from git): 
In `Cargo.toml`
notify = { version = "5.0.0-pre.2", optional = true }


#### What you did (as detailed as you can)

On the [bevy](https://github.com/bevyengine/bevy) github page, our CI builds are failing due to a recent update within the `fsevent-sys` crate.
See CI failures [here](https://github.com/bevyengine/bevy/pull/2149/checks?check_run_id=2567791064);

#### What you expected

No CI failure on a simple minor release version change.

#### What happened

CI failed due to: 
```
error[E0308]: mismatched types
   --> /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/notify-5.0.0-pre.7/src/fsevent.rs:361:17
    |
361 |                 callback,
    |                 ^^^^^^^^ expected `u32`, found enum `c_void`
    |
    = note: expected fn pointer `extern "C" fn(_, _, _, _, *const u32, *const u64)`
                  found fn item `extern "C" fn(_, _, _, _, *const c_void, *const c_void) {callback}`
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -34,8 +34,8 @@ inotify = { version = "0.9", default-features = false }
 mio = { version = "0.7.7", features = ["os-ext"] }
 
 [target.'cfg(target_os="macos")'.dependencies]
-fsevent = "2.0.1"
-fsevent-sys = "3"
+fsevent = "~2.0.1"
+fsevent-sys = "~3.1"
 
 [target.'cfg(windows)'.dependencies]
 winapi = { version = "0.3.8", features = ["fileapi", "handleapi", "ioapiset", "minwinbase", "synchapi", "winbase", "winnt"] }

